dev/benchmarks/macrobenchmarks/lib/src/opacity_peephole.dart

```diff
@@ -0,0 +1,303 @@
+  // The success of this case would depend on the sophistication of the non-overlapping tests.
+  OpacityPeepholeCase(
+    route: kOpacityPeepholeOpacityOfGridRouteName,
+    name: 'Opacity of Grid',
+    builder: (double v) {
+      double rowV = v;
+      double colV = v;
+      return Opacity(
+        opacity: _opacity(v),
+        child: SizedBox(
+          width: 300,
+          height: 400,
+          child: Stack(
+            children: <Widget>[
+              for (int i = 0; i < 10; i++, rowV = 1 - rowV, colV = rowV)
+                for (int j = 0; j < 7; j++, colV = 1 - colV)
+                  Positioned.fromRect(
+                    rect: Rect.fromLTWH(j * 40 + 5, i * 40 + 5, 30, 30),
+                    // RepaintBoundary here to avoid combining the 70 children into a single Picture
+                    child: RepaintBoundary(
+                      child: Container(
+                        color: Color.fromARGB(255, _red(colV), _green(colV), _blue(colV)),
+                      ),
+                    ),
+                  ),
+            ],
+          ),
+        ),
+      );
+    },
+  ),
+  // tests if an Opacity can hand its value down to a Column of non-overlapping rows of non-overlapping simple children.
+  // This test only requires linear non-overlapping tests to succeed.
+  OpacityPeepholeCase(
+    route: kOpacityPeepholeOpacityOfColOfRowsRouteName,
+    name: 'Opacity of Column of Rows',
+    builder: (double v) {
+      double rowV = v;
+      double colV = v;
+      return Opacity(
+        opacity: _opacity(v),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            for (int i = 0; i < 10; i++, rowV = 1 - rowV, colV = rowV)
+              Padding(
+                padding: const EdgeInsets.only(top: 5, bottom: 5),
+                // RepaintBoundary here to separate each row into a separate layer child
+                child: RepaintBoundary(
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: <Widget>[
+                      for (int j = 0; j < 7; j++, colV = 1 - colV)
+                        Padding(
+                          padding: const EdgeInsets.only(left: 5, right: 5),
+                          // RepaintBoundary here to prevent the row children combining into a single Picture
+                          child: RepaintBoundary(
+                            child: Container(
+                              width: 30,
+                              height: 30,
+                              color: Color.fromARGB(255, _red(colV), _green(colV), _blue(colV)),
+                            ),
+                          ),
+                        ),
+                    ],
+                  ),
+                ),
+              ),
+          ],
+        ),
+      );
+    },
+  ),
+];
```

Comment:
These look good, but a very common use case here is the `FadeTransition` widget. Do you think it'd be worth adding here, or perhaps in its own benchmark?

In particular, something like a `FadeTransition(opacity: ..., child: Text('Hello, world'))` seems good.

Reply:
Done.